ui/app/ducks/alerts/unconnected-account.js

```diff
@@ -0,0 +1,101 @@
+import { createSlice } from '@reduxjs/toolkit'
+import { captureException } from '@sentry/browser'
+
+import actionConstants from '../../store/actionConstants'
+import { addPermittedAccount } from '../../store/actions'
+import { getOriginOfCurrentTab } from '../../selectors/selectors'
+
+// Constants
+
+export const ALERT_STATE = {
+  CLOSED: 'CLOSED',
+  ERROR: 'ERROR',
+  LOADING: 'LOADING',
+  OPEN: 'OPEN',
+}
+
+const name = 'unconnectedAccount'
+
+const initialState = {
+  state: ALERT_STATE.CLOSED,
+  address: null,
+}
+
+// Slice (reducer plus auto-generated actions and action creators)
+
+const slice = createSlice({
+  name,
+  initialState,
+  reducers: {
+    connectAccountFailed: (state) => {
+      state.state = ALERT_STATE.ERROR
+    },
+    connectAccountRequested: (state) => {
+      state.state = ALERT_STATE.LOADING
+    },
+    connectAccountSucceeded: (state) => {
+      state.state = ALERT_STATE.CLOSED
+      state.address = null
+    },
+    dismissAlert: (state) => {
+      state.state = ALERT_STATE.CLOSED
+      state.address = null
+    },
+    switchedToUnconnectedAccount: (state, action) => {
+      state.state = ALERT_STATE.OPEN
+      state.address = action.payload
+    },
+  },
+  extraReducers: {
+    [actionConstants.UPDATE_METAMASK_STATE]: (state, action) => {
+      // lose the alert if the account is switched while it's open
```

Comment:
Nit: typo, this should say `close`, not `lose`
```suggestion
      // close the alert if the account is switched while it's open
```